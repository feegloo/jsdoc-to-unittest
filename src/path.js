export default ([base, ...parts], isFunction) => {
  let path = base;
  parts.forEach((part) => {
    if (Number.isNaN(+part) || typeof part !== 'string') {
      path += `.${part}`;
    } else {
      path += `[${part}]`;
    }
  });
  if (isFunction) {
    path += '()';
  }

  return path;
};
