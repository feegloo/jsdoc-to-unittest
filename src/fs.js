import fs from 'fs';
import path from 'path';
import util from 'util';

const writeFileAsync = util.promisify(fs.writeFile);
const readFileAsync = util.promisify(fs.readFile);

export const readFile = filename => readFileAsync(path.resolve(filename), 'utf8');
export const writeFile = (filename, content) => writeFileAsync(filename, content);

export const toStdout = (output) => {
  process.stdout.write(output);
  return true;
};
