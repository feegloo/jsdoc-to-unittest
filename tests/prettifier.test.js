import fs from 'fs';
import util from 'util';
import path from 'path';
import prettify from 'src/prettifier';

const readFileAsync = util.promisify(fs.readFile);

describe('Prettifier', () => {
  test('matches snapshot', async () => {
    const content = await readFileAsync(path.resolve(__dirname, './fixtures/bad-format.js'), 'utf-8');
    expect(await prettify(content)).toMatchSnapshot();
  });
});
