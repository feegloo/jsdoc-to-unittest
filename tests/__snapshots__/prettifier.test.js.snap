// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prettifier matches snapshot 1`] = `
"function abc() {
  let x = 2,
    y = 10;
  if (true) {
  }
  return 2, 10;
}
"
`;
